GsaGH/Components/0_Model/Units.cs

```diff
@@ -0,0 +1,224 @@
+        GH_ParamAccess.item);
+      pManager.AddTextParameter("Mass", "M", "Set Mass Unit for this GSA Model",
+        GH_ParamAccess.item);
+      pManager.AddTextParameter("Stress", "S", "Set Stress Unit for this GSA Model",
+        GH_ParamAccess.item);
+      pManager.AddTextParameter("Time Long", "TL", "Set Time Long Unit for this GSA Model",
+        GH_ParamAccess.item);
+      pManager.AddTextParameter("Time Medium", "TM", "Set Time Medium Unitfor this GSA Model",
+        GH_ParamAccess.item);
+      pManager.AddTextParameter("Time Short", "TS", "Set Time Short Unit for this GSA Model",
+        GH_ParamAccess.item);
+      pManager.AddTextParameter("Velocity", "V", "Set Velocity Unit for this GSA Model",
+        GH_ParamAccess.item);
+      for (int i = 1; i < pManager.ParamCount; i++) {
+        pManager[i].Optional = true;
+      }
+    }
+
+    protected override void RegisterOutputParams(GH_OutputParamManager pManager) {
+      pManager.AddParameter(new GsaModelParameter());
+      pManager.AddTextParameter("Acceleration", "Acc", "Get Acceleration Unit for this GSA Model",
+        GH_ParamAccess.item);
+      pManager.AddTextParameter("Angle", "Ang", "Get Angle Unit for this GSA Model",
+        GH_ParamAccess.item);
+      pManager.AddTextParameter("Energy", "E", "Get Energy Unit for this GSA Model",
+        GH_ParamAccess.item);
+      pManager.AddTextParameter("Force", "F", "Get Force Unit for this GSA Model",
+        GH_ParamAccess.item);
+      pManager.AddTextParameter("Geometry", "LL", "Get 'Length - large' Unit for this GSA Model",
+        GH_ParamAccess.item);
+      pManager.AddTextParameter("Property Dimension", "LP", "Get 'Length - sections' Unit for this GSA Model",
+        GH_ParamAccess.item);
+      pManager.AddTextParameter("Displacement", "LS", "Get 'Length - small' Unit for this GSA Model",
+        GH_ParamAccess.item);
+      pManager.AddTextParameter("Mass", "M", "Get Mass Unit for this GSA Model",
+        GH_ParamAccess.item);
+      pManager.AddTextParameter("Stress", "S", "Get Stress Unit for this GSA Model",
+        GH_ParamAccess.item);
+      pManager.AddTextParameter("Time Long", "TL", "Get Time Long Unit for this GSA Model",
+        GH_ParamAccess.item);
+      pManager.AddTextParameter("Time Medium", "TM", "Get Time Medium Unitfor this GSA Model",
+        GH_ParamAccess.item);
+      pManager.AddTextParameter("Time Short", "TS", "Get Time Short Unit for this GSA Model",
+        GH_ParamAccess.item);
+      pManager.AddTextParameter("Velocity", "V", "Get Velocity Unit for this GSA Model",
+        GH_ParamAccess.item);
+    }
+
+    protected override void SolveInstance(IGH_DataAccess da) {
+      GsaModelGoo gooModel = null;
+      GsaModel model = null;
+      int i = 0;
+      if (da.GetData(i++, ref gooModel)) {
+        model = gooModel.Value.Duplicate(true);
+      }
+      GsaAPI.UiUnits units = model.Units;
```

Comment:
if model will be null here we will have error

Reply:
which will be caught by grasshopper and exposed to the user